src/app/components/action-button.ts

```diff
@@ -0,0 +1,18 @@
+/// <reference path="../../../typings/tsd.d.ts" />
+
+import {Component, View, ElementRef} from 'angular2/angular2';
+
+@Component({
+	selector: 'action-button'
+})
+@View({
+	template: `<button (click)="doAction()">Action!</button>`
+})
+export class ActionButton {
+	constructor(public el:ElementRef) {
+		
+	}
+	doAction(){
+		this.el.domElement.dispatchEvent(new Event('yummy'));
```

Comment:
Whoa, thanks! :)

-----Original Message-----
From: "PatrickJS" <notifications@github.com>
Sent: ‎6/‎11/‎2015 11:22 PM
To: "angular-class/angular2-webpack-starter" <angular2-webpack-starter@noreply.github.com>
Cc: "Ryan Graff" <ryantgraff@resistdesign.com>
Subject: Re: [angular2-webpack-starter] feat(todo item) Add an action buttonwith a custom yummy event. (#23)

In src/app/components/action-button.ts:
> +/// <reference path="../../../typings/tsd.d.ts" />
> +
> +import {Component, View, ElementRef} from 'angular2/angular2';
> +
> +@Component({
> +	selector: 'action-button'
> +})
> +@View({
> +	template: `<button (click)="doAction()">Action!</button>`
> +})
> +export class ActionButton {
> +	constructor(public el:ElementRef) {
> +		
> +	}
> +	doAction(){
> +		this.el.domElement.dispatchEvent(new Event('yummy'));
rather than using native events you should use angular's version.
@Component({
  selector: 'action-button'
  events: ['click']
})
@View({
  template: `<button (click)="doAction($event)">Action!</button>`
})
export class ActionButton {
    constructor(public el:ElementRef) {

    }
    doAction(event) {
        this.click.next({ data: 'yolo', event: event })
  }
}—
Reply to this email directly or view it on GitHub.